internal/state/merkle/serialization_utils_test.go

```diff
@@ -10,25 +12,63 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-// TestGenerateStateKey verifies that the state key generation works as expected.
-func TestGenerateStateKey(t *testing.T) {
-	// Test with i and serviceId
-	i := uint8(1)
-	serviceId := block.ServiceId(100)
-
-	// Generate the state key
-	stateKey := generateStateKey(i, serviceId)
-
-	// Verify the length is 32 bytes
-	assert.Equal(t, 32, len(stateKey))
-
-	// Verify that the first byte matches i
-	assert.Equal(t, i, stateKey[0])
+// TestGenerateStateKeyInterleavedBasic verifies that the state key generation works as expected.
+func TestGenerateStateKeyInterleavedBasic(t *testing.T) {
+	tests := []struct {
+		name      string
+		i         uint8
+		serviceId block.ServiceId
+	}{
+		{
+			name:      "basic case",
+			i:         1,
+			serviceId: 100,
+		},
+		{
+			name:      "max values",
+			i:         255,
+			serviceId: block.ServiceId(math.MaxUint32),
+		},
+		{
+			name:      "zero values",
+			i:         0,
+			serviceId: 0,
+		},
+	}
 
-	// Optionally, verify that the encoded serviceId is in the key
-	expectedEncodedServiceId := make([]byte, 4)
-	binary.BigEndian.PutUint32(expectedEncodedServiceId, uint32(serviceId))
-	assert.Equal(t, expectedEncodedServiceId, stateKey[1:5])
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Generate the state key
+			stateKey := generateStateKeyInterleavedBasic(tt.i, tt.serviceId)
+
+			// Convert serviceId to bytes for verification
+			serviceIdBytes := make([]byte, 4)
+			binary.BigEndian.PutUint32(serviceIdBytes, uint32(tt.serviceId))
```

Comment:
Is there any reason why you use big endian encoding? I think you should use jam integer encoding here (little endian)

Reply:
If you see the implementation of generateStateKeyInterleavedBasic, we need that ordering. With Little Endian won't pass

Reply:
Agreed that the implmentation should be different. Updated